pkg/api/types.go

```diff
@@ -28,6 +28,53 @@ type Project struct {
 	UpdatedTime time.Time `bson:"updatedTime,omitempty" json:"updatedTime,omitempty" description:"updated time of the project"`
 }
 
+// PipelineRecord is all information about some pipeline building history.
+type PipelineRecord struct {
+	ID          string       `bson:"_id,omitempty" json:"id,omitempty" description:"id of the pipelineRecord"`
+	PipelineID  string       `bson:"pipelineID,omitempty" json:"pipelineID,omitempty" description:"id of the related pipeline which the pipelineRecord belongs to"`
+	VersionID   string       `bson:"versionID,omitempty" json:"versionID,omitempty" description:"id of the related version which the pipelineRecord belongs to"`
+	Trigger     string       `bson:"trigger,omitempty" json:"trigger,omitempty" description:"trigger of the pipelineRecord"`
+	StageStatus *StageStatus `bson:"stageStatus,omitempty" json:"stageStatus,omitempty" description:"status of the latest stage"`
+	Status      Status       `bson:"status,omitempty" json:"status,omitempty" description:"status of the pipelineRecord"`
+	StartTime   time.Time    `bson:"startTime,omitempty" json:"startTime,omitempty" description:"startTime of the pipelineRecord"`
+	EndTime     time.Time    `bson:"endTime,omitempty" json:"endTime,omitempty" description:"endTime of the pipelineRecord"`
```

Comment:
Variable names should not be used in description and comments, it is better to change to `end time of the pipeline record`. Please fix the same error in other places at the same time.